f5lbaasdriver/v2/bigip/service_builder.py

```diff
@@ -277,20 +281,26 @@ def _populate_member_network(self, context, member, network):
         member['vxlan_vteps'] = []
         member['gre_vteps'] = []
 
-        if 'provider:network_type' in network:
-            net_type = network['provider:network_type']
-            if net_type == 'vxlan':
-                if 'binding:host_id' in member['port']:
-                    host = member['port']['binding:host_id']
-                    member['vxlan_vteps'] = self._get_endpoints(
-                        context, 'vxlan', host)
-            if net_type == 'gre':
-                if 'binding:host_id' in member['port']:
-                    host = member['port']['binding:host_id']
-                    member['gre_vteps'] = self._get_endpoints(
-                        context, 'gre', host)
-        if 'provider:network_type' not in network:
-            network['provider:network_type'] = 'undefined'
+        agent_config = {}
```

Comment:
Should not be an empty dict.  Save the agent_config obtained in build() on line 91 as an instance attribute to share with this method.  This will be required to address my comment in get_network_segment to look generically for 'supported network type' and not precisely for 'opflex'.